Type mismatches for jnp.concatenate tests

This may be working as intended, but I disabled these tests for the moment:

```
====================================================================================================== FAILURES ======================================================================================================
________________________________________________________________________________________ LaxBackedNumpyTests.testConcatenate4 ________________________________________________________________________________________
[gw0] linux -- Python 3.10.0 /usr/local/google/home/phawkins/.pyenv/versions/3.10.0/envs/py310/bin/python3.10

self = <lax_numpy_test.LaxBackedNumpyTests testMethod=testConcatenate4>, axis = 0, dtype = None, base_shape = (3, 4), arg_dtypes = (<class 'numpy.float16'>, <class 'numpy.int16'>, <class 'numpy.int16'>)

    @jtu.sample_product(
      [dict(base_shape=base_shape, axis=axis)
        for base_shape in [(4,), (3, 4), (2, 3, 4)]
        for axis in range(-len(base_shape)+1, len(base_shape))
      ],
      arg_dtypes=[
        arg_dtypes
        for num_arrs in [3]
        for arg_dtypes in itertools.combinations_with_replacement(default_dtypes, num_arrs)
      ],
      dtype=[None] + default_dtypes,
    )
    def testConcatenate(self, axis, dtype, base_shape, arg_dtypes):
      rng = jtu.rand_default(self.rng())
      wrapped_axis = axis % len(base_shape)
      shapes = [base_shape[:wrapped_axis] + (size,) + base_shape[wrapped_axis+1:]
                for size, _ in zip(itertools.cycle([3, 1, 4]), arg_dtypes)]
      def np_fun(*args):
        args = [x if x.dtype != jnp.bfloat16 else x.astype(np.float32)
                for x in args]
        return np.concatenate(args, axis=axis, dtype=dtype, casting='unsafe')
      jnp_fun = lambda *args: jnp.concatenate(args, axis=axis, dtype=dtype)

      def args_maker():
        return [rng(shape, dtype) for shape, dtype in zip(shapes, arg_dtypes)]

      with jtu.strict_promotion_if_dtypes_match(arg_dtypes):
        # # TODO: tests fail with bfloat16/float16 inputs mixed with integers.
        # if np.float16 not in arg_dtypes and dtypes.bfloat16 not in arg_dtypes:
>       self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker)

tests/lax_numpy_test.py:1462:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
jax/_src/test_util.py:918: in _CheckAgainstNumpy
    self.assertAllClose(numpy_ans, lax_ans, check_dtypes=check_dtypes,
jax/_src/test_util.py:842: in assertAllClose
    self.assertDtypesMatch(x, y, canonicalize_dtypes=canonicalize_dtypes)
jax/_src/test_util.py:817: in assertDtypesMatch
    self.assertEqual(_dtypes.canonicalize_dtype(_dtype(x)),
E   AssertionError: dtype('float32') != dtype('float16')
________________________________________________________________________________________ LaxBackedNumpyTests.testConcatenate8 ________________________________________________________________________________________
[gw0] linux -- Python 3.10.0 /usr/local/google/home/phawkins/.pyenv/versions/3.10.0/envs/py310/bin/python3.10

self = <lax_numpy_test.LaxBackedNumpyTests testMethod=testConcatenate8>, axis = -2, dtype = None, base_shape = (2, 3, 4), arg_dtypes = (<class 'bfloat16'>, <class 'numpy.int8'>, <class 'numpy.int8'>)

    @jtu.sample_product(
      [dict(base_shape=base_shape, axis=axis)
        for base_shape in [(4,), (3, 4), (2, 3, 4)]
        for axis in range(-len(base_shape)+1, len(base_shape))
      ],
      arg_dtypes=[
        arg_dtypes
        for num_arrs in [3]
        for arg_dtypes in itertools.combinations_with_replacement(default_dtypes, num_arrs)
      ],
      dtype=[None] + default_dtypes,
    )
    def testConcatenate(self, axis, dtype, base_shape, arg_dtypes):
      rng = jtu.rand_default(self.rng())
      wrapped_axis = axis % len(base_shape)
      shapes = [base_shape[:wrapped_axis] + (size,) + base_shape[wrapped_axis+1:]
                for size, _ in zip(itertools.cycle([3, 1, 4]), arg_dtypes)]
      def np_fun(*args):
        args = [x if x.dtype != jnp.bfloat16 else x.astype(np.float32)
                for x in args]
        return np.concatenate(args, axis=axis, dtype=dtype, casting='unsafe')
      jnp_fun = lambda *args: jnp.concatenate(args, axis=axis, dtype=dtype)

      def args_maker():
        return [rng(shape, dtype) for shape, dtype in zip(shapes, arg_dtypes)]

      with jtu.strict_promotion_if_dtypes_match(arg_dtypes):
        # # TODO: tests fail with bfloat16/float16 inputs mixed with integers.
        # if np.float16 not in arg_dtypes and dtypes.bfloat16 not in arg_dtypes:
>       self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker)

tests/lax_numpy_test.py:1462:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
jax/_src/test_util.py:918: in _CheckAgainstNumpy
    self.assertAllClose(numpy_ans, lax_ans, check_dtypes=check_dtypes,
jax/_src/test_util.py:842: in assertAllClose
    self.assertDtypesMatch(x, y, canonicalize_dtypes=canonicalize_dtypes)
jax/_src/test_util.py:817: in assertDtypesMatch
    self.assertEqual(_dtypes.canonicalize_dtype(_dtype(x)),
E   AssertionError: dtype('float32') != dtype(bfloat16)
============================================================================================== short test summary info ===============================================================================================
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testConcatenate4 - AssertionError: dtype('float32') != dtype('float16')
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testConcatenate8 - AssertionError: dtype('float32') != dtype(bfloat16)
```

--- a/tests/lax_numpy_test.py
+++ b/tests/lax_numpy_test.py
@@ -1447,7 +1447,9 @@ class LaxBackedNumpyTests(jtu.JaxTestCase):
     wrapped_axis = axis % len(base_shape)
     shapes = [base_shape[:wrapped_axis] + (size,) + base_shape[wrapped_axis+1:]
               for size, _ in zip(itertools.cycle([3, 1, 4]), arg_dtypes)]
-    def np_fun(*args):
+    @_promote_like_jnp
+    def np_fun(*args, dtype=dtype):
+      dtype = dtype or args[0].dtype
       args = [x if x.dtype != jnp.bfloat16 else x.astype(np.float32)
               for x in args]
       return np.concatenate(args, axis=axis, dtype=dtype, casting='unsafe')
@@ -1457,9 +1459,7 @@ class LaxBackedNumpyTests(jtu.JaxTestCase):
       return [rng(shape, dtype) for shape, dtype in zip(shapes, arg_dtypes)]
 
     with jtu.strict_promotion_if_dtypes_match(arg_dtypes):
-      # TODO: tests fail with bfloat16/float16 inputs mixed with integers.
-      if np.float16 not in arg_dtypes and dtypes.bfloat16 not in arg_dtypes:
-        self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker)
+      self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker)
       self._CompileAndCheck(jnp_fun, args_maker)
 
   @jtu.sample_product(

